### PR TITLE
backend/rv64: Add minimal toy implementation enough to execute LSLS

### PR DESCRIPTION
**Pull request by ksco**

### CODE REVIEW

File: src/dynarmic/backend/riscv64/reg_alloc.cpp [hunk 1]
```diff
@@ -9,10 +9,16 @@
...
 #include <mcl/assert.hpp>
+#include <mcl/mp/metavalue/lift_value.hpp>
 #include <mcl/stdint.hpp>
 
+#include "dynarmic/common/always_false.h"
+
 namespace Dynarmic::Backend::RV64 {
 
+// TODO: We should really move this to biscuit.
+void Mov64(biscuit::Assembler& as, biscuit::GPR rd, u64 imm);
```
Comment (ksco) [inline src/dynarmic/backend/riscv64/reg_alloc.cpp:20]: I'll do this next PR.

**Comment by merryhime** [comment]
Removed 2x comments because I understand the previous bug now.

File: src/dynarmic/backend/riscv64/a32_address_space.cpp [hunk 2]
```diff
@@ -78,21 +78,21 @@ void A32AddressSpace::EmitPrelude() {
...
-        as.FSD(FPR{i}, 32 + i * 8, sp);
+        as.FSD(FPR{i}, (32 + i) * 8, sp);
     }
 
     as.ADDI(Xstate, a1, 0);
     as.ADDI(Xhalt, a2, 0);
     as.JALR(x0, 0, a0);
 
-    prelude_info.return_from_run_code = GetCursorPtr<CodePtr>();
+    prelude_info.return_from_run_code = GetCursorPtr<CodePtr>() + 4;
```
[the remark this answers is not in the document]
Comment (ksco) [reply src/dynarmic/backend/riscv64/a32_address_space.cpp]: Sorry, I've fixed the real cause: https://github.com/merryhime/dynarmic/compare/2f8227e..db3bc4b. (The `SLLI` -> `SLLI64` fix is not related to this and the function will be deleted in the next PR anyway.)

**Comment by ksco** [comment]
@merryhime Hi, there is no rush, but is there more needs to be done for this PR? I'm ready for the next PR ;)

**Pull request merged**